src/main/kotlin/com/jetbrains/micropython/repl/StopReplBeforeRunTask.kt

```diff
@@ -0,0 +1,47 @@
+package com.jetbrains.micropython.repl
+
+import com.intellij.execution.BeforeRunTask
+import com.intellij.execution.BeforeRunTaskProvider
+import com.intellij.execution.configurations.RunConfiguration
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.util.Key
+import com.jetbrains.micropython.run.MicroPythonRunConfiguration
+import com.jetbrains.micropython.settings.MicroPythonFacetType
+
+class StopReplBeforeRunTask : BeforeRunTask<StopReplBeforeRunTask>(StopReplBeforeRunTaskProvider.ID) {
+    init {
+        isEnabled = true
+    }
+}
+
+class StopReplBeforeRunTaskProvider : BeforeRunTaskProvider<StopReplBeforeRunTask>() {
```

Comment:
This run before task might be unnecessary. See my comments to `RunStateWrapper`.